src/Microsoft.Extensions.ProjectModel.Sources/MsBuild/MsBuildProjectDependencyProvider.cs

```diff
@@ -0,0 +1,155 @@
+            string name = string.Empty;
+
+            resolvedPath = item.EvaluatedInclude;
+
+            name = resolvedPath ?? Path.GetFileNameWithoutExtension(resolvedPath);
+
+            foreach (var m in item.Metadata)
+            {
+                if (ItemSpecKey.Equals(m.Name, StringComparison.OrdinalIgnoreCase))
+                {
+                    itemSpec = m.EvaluatedValue;
+                    continue;
+                }
+
+                if (NugetSourceTypeKey.Equals(m.Name, StringComparison.OrdinalIgnoreCase))
+                {
+                    nugetPackageSource = m.EvaluatedValue;
+                    continue;
+                }
+
+                if (VersionKey.Equals(m.Name, StringComparison.OrdinalIgnoreCase))
+                {
+                    version = m.EvaluatedValue;
+                }
+            }
+
+            var isFromNugetPackage = "Package".Equals(nugetPackageSource, StringComparison.OrdinalIgnoreCase);
+            return new ResolvedReference(name, version, resolvedPath, isFromNugetPackage);
+        }
+
+        private static DependencyDescription CreateDependencyDescriptionFromItem(ProjectItemInstance item)
+        {
+            var version = item.GetMetadataValue("Version");
+            var path = item.GetMetadataValue("Path");
+            var type = item.GetMetadataValue("Type");
+            var resolved = item.GetMetadataValue("Resolved");
+            var itemSpec = item.EvaluatedInclude;
+
+            // For type == Target, we do not get Name in the metadata. This is a special node where the dependencies are 
+            // the direct dependencies of the project.
+
+            var name = item.HasMetadata("Name") ? item.GetMetadataValue("Name") : itemSpec;
+
+            if (string.IsNullOrEmpty(name))
+            {
+                return null;
+            }
+
+            bool isResolved = false;
+            isResolved = bool.TryParse(resolved, out isResolved) ? isResolved : false;
+            var framework = itemSpec.Split(new char[] { '/' }, StringSplitOptions.RemoveEmptyEntries).First();
+
+            return new DependencyDescription(name, version, path, framework, type, isResolved);
+        }
+
+        private static void PopulateDependencies(Dictionary<string, DependencyDescription> dependencies, ICollection<ProjectItemInstance> items)
+        {
+            foreach (var item in items)
+            {
+                var depSpecs = item.GetMetadataValue("Dependencies")
+                    ?.Split(new char[] { ';' }, StringSplitOptions.RemoveEmptyEntries);
+                DependencyDescription current = null;
+                if (depSpecs == null || !dependencies.TryGetValue(item.EvaluatedInclude, out current))
+                {
+                    return;
+                }
+
+                foreach (var depSpec in depSpecs)
+                {
+                    var spec = item.EvaluatedInclude.Split('/').FirstOrDefault()+ "/" + depSpec;
```

Comment:
- Move the prefix here outside of the foreach loop since the prefix doesn't change.
- Use IndexOf and substring instead of Split.
- `$"{prefix}/{depSpec}"` is your friend